### DISCUSSION TITLE
Importing a component that needs server-only

### DISCUSSION THREAD

**Discussion by haris-sattar** [vercel/next.js#66015] [Help]
### Summary

I am using T3 to build a website. I have created my database schema and am trying to access some basic information from one of the tables. I created a router on the server and I am trying to call it on the client side, but no matter which file I put it in, it says "You're importing a component that needs server-only".  

### Additional information

```js
You're importing a component that needs server-only. That only works in a Server Component which is not supported in the pages/ directory.

import { z } from "zod";

import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";

export const postRouter = createTRPCRouter({
 
  getLatest: publicProcedure.query(({ ctx }) => {
    return ctx.db.post.findFirst({
      orderBy: { createdAt: "desc" },
    });
  }),
});

In a client side file, I tried running this in multiple files:
const a = api.post.getLatest();
console.log(a);

I tried using await and that didn't work either. Any advice?
```

**Comment by hasan-mohammad-bd** [comment]
You're trying to use (createTRPCRouter) is only compatible with Server Components, which are not supported in the pages/ directory. 

To fix this issue, you'll need to move the code that uses createTRPCRouter to a server-side file, such as an API route in your pages/api/ directory. Then, you can call that API route from your client-side code using the fetch API
